File: objective-c/Side/01_raindrops/Raindrops.m
#import "Raindrops.h"

@interface Raindrops ()

@property (nonatomic) int number;

@end

@implementation Raindrops

#pragma mark - Init

- (id) initWithNumber:(int)number {
    self = [super init];

    self.number = number;

    return self;
}

#pragma mark - Object Logic

- (NSString *) sounds {
    NSString *pling = (self.number % 3 == 0) ? @"Pling" : @"";
    NSString *plang = (self.number % 5 == 0) ? @"Plang" : @"";
    NSString *plong = (self.number % 7 == 0) ? @"Plong" : @"";

    NSString *joined = [@[pling, plang, plong] componentsJoinedByString:@""];

    return (joined.length == 0) ? [NSString stringWithFormat:@"%d", self.number] : joined;
}

@end

File: swift/Side/03_raindrops/Sources/Raindrops/Raindrops.swift
//Solution goes in Sources

struct Raindrops {

    let drop: Int

    var sounds: String {
        let pling = (self.drop % 3 == 0) ? "Pling" : ""
        let plang = (self.drop % 5 == 0) ? "Plang" : ""
        let plong = (self.drop % 7 == 0) ? "Plong" : ""

        let joined = [pling, plang, plong].joined()

        if joined.isEmpty {
            return "\(self.drop)"
        } else {
            return joined
        }
    }

    init(_ drop: Int) {
        self.drop = drop
    }

}
